hms-app/src/components/admin-tools/UserDetails.tsx

```diff
@@ -0,0 +1,220 @@
+import { deleteUser, getUserDetails } from '../../actions/UserActions'
+import React, { useEffect, useState } from 'react'
+import { User } from '../../common/types'
+import {
+  Accordion,
+  Badge,
+  Button,
+  Card,
+  createStyles,
+  Group,
+  Modal,
+  Text,
+  useMantineTheme,
+} from '@mantine/core'
+import HackathonForm from './HackathonForm'
+import UserForm from './UserForm'
+
+const useStyles = createStyles((theme) => ({
+  card: {
+    backgroundColor:
+      theme.colorScheme === 'dark' ? theme.colors.dark[7] : theme.white,
+  },
+  section: {
+    borderBottom: `1px solid ${
+      theme.colorScheme === 'dark' ? theme.colors.dark[4] : theme.colors.gray[3]
+    }`,
+    paddingLeft: theme.spacing.md,
+    paddingRight: theme.spacing.md,
+    paddingBottom: theme.spacing.md,
+    fontSize: theme.fontSizes.md,
+    fontWeight: 500,
+  },
+  label: {
+    textTransform: 'uppercase',
+    fontSize: theme.fontSizes.xs,
+    fontWeight: 700,
+  },
+}))
+
+export default function UserDetails(props: { userID: string }) {
+  const theme = useMantineTheme()
+  const { classes } = useStyles()
+  const { userID } = props
+  const [deleteModalOpened, setDeleteModalOpened] = useState(false)
+  const [editModalOpened, setEditModalOpened] = useState(false)
+  const [isUserError, setIsUserError] = useState(false)
+  const [isUserLoading, setIsUserLoading] = useState(true)
+  const [user, setUser] = useState({
+    id: 'string',
+    lastName: 'string',
+    firstName: 'string',
+    emailAddress: 'string',
+    roles: [],
+    skills: [],
+    imageUrl: 'string',
+    creationDate: new Date(),
+  } as User)
+
+  const loadSelectedUser = () => {
+    getUserDetails(userID).then(
+      (data) => {
+        setUser({
+          id: data.id,
+          lastName: data.lastName,
+          firstName: data.firstName,
+          emailAddress: data.emailAddress,
+          roles: data.roles,
+          skills: data.skills,
+          imageUrl: data.imageUrl,
+          creationDate: data.creationDate,
+        })
```

Comment:
```suggestion
        setUser(data)
```

Reply:
it might complain about this though.  To fix it you would need to make it so that `getUserDetails()` returns `Promise<User>`